src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,169 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int VALID_AGE = 18;
+    private static final String VALID_LOGIN = "validlogin";
+    private static final String VALID_PASSWORD = "validPassword";
+    private static final int ZERO_AGE = 0;
+    private static final int NEGATIVE_AGE = -15;
+    private static final int SMALL_AGE = 15;
+    private static final String SMALL_LOGIN = "bad";
+    private static final String EMPTY_LOGIN = "";
+    private static final String NULL_LOGIN = null;
+    private static final String FIVE_LETTERS_LOGIN = "worst";
+    private static final String SMALL_PASSWORD = "bad";
+    private static final String EMPTY_PASSWORD = "";
+    private static final String NULL_PASSWORD = null;
+    private static final String FIVE_LETTERS_PASSWORD = "worst";
+    private RegistrationService registrationService;
+
+    @BeforeEach
+    void setUp() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    User validUserSetUp() {
+        User actualUser = new User();
+        actualUser.setAge(VALID_AGE);
+        actualUser.setLogin(VALID_LOGIN);
+        actualUser.setPassword(VALID_PASSWORD);
+        return actualUser;
+    }
+
+    @Test
+    void register_validUser_isOk() {
+        User actualUser = validUserSetUp();
+        Storage.people.clear();
+        assertNotNull(registrationService.register(actualUser));
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        User actualUser = null;
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(actualUser);
+        });
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
```

Comment:
It is better to first have tesks wich is ok than notOk, wich will be grouped (first all tests with invalid login than password etc)